AirCasting/MobilePeripheralSessionManager.swift

```diff
@@ -0,0 +1,73 @@
+// Created by Lunar on 07/06/2021.
+//
+
+import Foundation
+import CoreBluetooth
+
+class MobilePeripheralSessionManager {
+    private let measurementStreamStorage: MeasurementStreamStorage
+    private lazy var locationProvider = LocationProvider()
```

Comment:
⚠️ Seams like `LocationProvider` has `startUpdatingLocation` used in `requestLocation` so triggered once it will never stop because it has no corresponding `stopRequestingLocation` without it the location is not updated and the `locationProvider.currentLocation?.coordinate` You are accessing maybe invalid.